File: 279-arithmeticOperationsForMatrix.py


def multiplyMatrix(a,b):
    R=len(a)
    assert R>0
    C=len(a[0])
    assert C==len(b)
    M=[[0 for _ in range(len(b[0]))] for _ in range(R) ]
    for i in range(R):
        for j in range(len(b[0])):
            for k in range(C):
                M[i][j]+=a[i][k]*b[k][j]
                
    return M

a=[[1,2,3],[4,5,6],[7,8,9]]
b=[[1,0,0],[0,1,0],[0,0,1]]
print(multiplyMatrix(a,b))

File: 279-arithmeticOperationsForMatrix.cpp
#include<iostream>
#include<vector>
#include "arithmeticOperationsOfMatrix.h"


DT multiplyMatrix(const DT& a, const DT& b)
{
    int R=a.size();
    int C=b[0].size();
    DT ans(R,std::vector<int>(C,0));  //there is inside std::vector<int>
    for(int i=0;i<R;++i)
    {
        for(int j=0;j<C;++j)
        {
            for(int k=0;k<C;++k)
            {
                ans[i][j]+=a[i][k]*b[k][j];
            }
        }
    }
    return ans;
}

int main()
{
    DT a{{1,2},{3,4}},b{{1,1},{1,1}};
    DT ret=multiplyMatrix(a,b);
    for(auto &d:ret)
    {
        std::cout<<'\n';
        for(auto i:d)
            std::cout<<i<<',';
    }
    return 0;
}